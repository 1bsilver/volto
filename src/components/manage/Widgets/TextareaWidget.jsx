/**
 * TextareaWidget component.
 * @module components/manage/Widgets/TextareaWidget
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Form, Label, TextArea } from 'semantic-ui-react';

/**
 * TextareaWidget component class.
 * @function TextareaWidget
 * @returns {string} Markup of the component.
 */
const TextareaWidget = ({
  id,
  title,
  required,
  description,
  error,
  value,
  onChange,
}) => (
  <Form.Field required={required} error={error}>
    <label htmlFor={`field-${id}`}>
      {title}
      {description && <span className="help">{description}</span>}
    </label>
    <TextArea
      id={`field-${id}`}
      name={id}
      value={value || ''}
      onChange={({ target }) =>
        onChange(id, target.value === '' ? undefined : target.value)}
    />
    {error && <Label basic color="red" pointing="below">{error}</Label>}
  </Form.Field>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
TextareaWidget.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

/**
 * Default properties.
 * @property {Object} defaultProps Default properties.
 * @static
 */
TextareaWidget.defaultProps = {
  description: null,
  required: false,
  error: null,
  value: null,
};

export default TextareaWidget;
