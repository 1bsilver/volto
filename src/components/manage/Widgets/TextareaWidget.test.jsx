import React from 'react';
import renderer from 'react-test-renderer';
import TextareaWidget from './TextareaWidget';

test('renders a textarea widget component', () => {
  const component = renderer.create(
    <TextareaWidget
      id="my-field"
      title="My field"
      onChange={() => {}}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
