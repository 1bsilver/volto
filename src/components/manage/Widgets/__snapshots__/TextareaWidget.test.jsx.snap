// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a textarea widget component 1`] = `
<div
  className="field"
>
  <label
    htmlFor="field-my-field"
  >
    My field
  </label>
  <textarea
    id="field-my-field"
    name="my-field"
    onChange={[Function]}
    value=""
  />
</div>
`;
